pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/upsertcompaction/UpsertCompactionTaskGenerator.java

```diff
@@ -218,10 +218,20 @@ public static SegmentSelectionResult processValidDocIdsMetadata(Map<String, Stri
         long totalDocs = validDocIdsMetadata.getTotalDocs();
         double invalidRecordPercent = ((double) totalInvalidDocs / totalDocs) * 100;
         if (totalInvalidDocs == totalDocs) {
+          LOGGER.info("Segment {} contains only invalid records, adding it to the deletion list", segmentName);
           segmentsForDeletion.add(segment.getSegmentName());
         } else if (invalidRecordPercent >= invalidRecordsThresholdPercent
             && totalInvalidDocs >= invalidRecordsThresholdCount) {
+          LOGGER.info("Segment {} contains {} invalid records out of {} total records "
+                          + "(count threshold: {}, percent threshold: {}), adding it to the compaction list",
+                  segmentName, totalInvalidDocs, totalDocs, invalidRecordsThresholdCount,
+                  invalidRecordsThresholdPercent);
           segmentsForCompaction.add(Pair.of(segment, totalInvalidDocs));
+        } else {
+          LOGGER.info("Segment {} contains {} invalid records out of {} total records "
+                          + "(count threshold: {}, percent threshold: {}), skipping it for compaction",
+                  segmentName, totalInvalidDocs, totalDocs, invalidRecordsThresholdCount,
+                  invalidRecordsThresholdPercent);
```

Comment:
Let's make these logs as DEBUG. We have 10000s of segments in a table and we run compaction every 5 mins for some of our tables. This might flood the logs.

Reply:
Another way is to collect some aggregated stats and log them in the end

Reply:
yep, i can switch to debug. i also added a summary log as well (how many compacted vs. deleted vs. skipped).